Seasons Defense/Assets/Scripts/TowerController.cs

```diff
@@ -4,24 +4,21 @@ public class TowerController : MonoBehaviour
 {
     public GameObject missilePrefab;
     public float offset;
+    
+    public void Fire() {
 
-    private void Update()
-    {
-        if (Input.GetMouseButtonDown(1))
-        {
-            Vector3 mousePosition = new Vector3(Input.mousePosition.x, Input.mousePosition.y, Camera.main.nearClipPlane);
-            Vector3 worldPosition = Camera.main.ScreenToWorldPoint(mousePosition);
-            worldPosition.z = transform.position.z;
-            GameObject missileInstance = Instantiate(missilePrefab, new Vector3(transform.position.x, transform.position.y + offset, 0), Quaternion.identity);
```

Comment:
This Vector3 will be the same each time a missile is fired from this tower. Can we move this Vector3 to a private variable initialized at Start() ?